Ebb advice

LenaEbb opened the pull request:
Hope never dies...

redrambles commented:
Hey @LenaEbb ! :)

It's true, hope never dies! And this is a good thing!
I would LOVE to say that we are all good to merge in this pull request, but there are some things we need to address. That's ok, we'll work through them together! :)

First - [let's take a look at this page](https://github.com/skillcrush/skillcrush-104/pull/1701/files). This is where we see what is contained within your pull request. We should be seeing three files here: 
1) Your advice page
2) A stylesheet. Either the main one (css/style.css) where you would have added your styles, or a custom one if you wanted to go that route.
3) The main index.html file, where you would have added a link to your advice page.

If you've checked out the link I pasted above, you can see that there are 10 files contained in your pull request instead of 3. 

Here's what we'd like you to do.

1) Pick one file for your advice
2) Pick one file for your CSS
3) Keep the index.html file but make sure that you've added only one additional link to it- the one to the advice file that you've decided to keep.

Delete the other 7 files.

Once you've done this, you'll want to add your changes with `git add -A` and then commit them `git commit -m "Lena's advice"` and then push them up once more `git push origin ebb-advice`, which will automatically update your pull request.

We'll go from there! Let us know if you have any questions, and keep working at it!!

LenaEbb commented:
Hey Ann, I really thought I understood this, but it confuses me more every minute. I wrote inline CSS on purpose so I only had one file to commit and make sure I do it right. I have no idea how all those other files got anywhere near me...Can you please explain that? Where did those files come from?  Also, I thought I deleted those previous files I wrote last week and changed the name of the new one, but everything was still there. I now deleted (at least I think I did) those files and there are now(at least I think so), only two files I need to commit, my html file with CSS in the <head> and the main html with the link of my new file ebb-advice18.html. I went to the terminal and followed your instructions, but when I tried to commit, another 10 or more files that have nothing to do with me showed and I'm getting an error message. I'm definitely watching the Git tutorial but what should I do for now? I haven't submitted another pull request, I need to figure this out first. Thank you!!! Waiting for further instructions lol, hope never dies

redrambles commented on index.html:
```diff
@@ -16,6 +16,10 @@ <h2>Advice for Newbies!</h2>
       <div class="all-advice-links">
         <div class="skillcrush">
+
+          <a href="advice-template-kk.html">KK's advice</a><br>
```
Erase this link

redrambles commented on index.html:
```diff
@@ -16,6 +16,10 @@ <h2>Advice for Newbies!</h2>
+
+          <a href="advice-template-kk.html">KK's advice</a><br>
+	  <a href="advice/ebb-advice18.html">Lena's Advice</a><br>
+          <a href="advice/advice-mimi.html">Mimi's Advice for Introverts</a><br>
```
Erase this link

redrambles commented on index.html:
```diff
@@ -16,6 +16,10 @@ <h2>Advice for Newbies!</h2>
         <div class="skillcrush">
+
+          <a href="advice-template-kk.html">KK's advice</a><br>
+	  <a href="advice/ebb-advice18.html">Lena's Advice</a><br>
```
Keep this one

redrambles commented:
Hey @LenaEbb !

Ok - we're getting much closer! I see we have two files now - but we really do want three, meaning that your CSS should be in a separate file. Your advice file shouldn't have any CSS inside of it - so just grab the CSS that you've placed there and put it either in a custom stylesheet and put it in the /css directory - or the main stylesheet (css/styles.css). Whichever one you choose, make sure to link to that file from your `advice/ebb-advice18.html` file. 

Once you've done that, we'll just need you to keep the link to your `advice/ebb-advice18.html` file in the main index.html - [but erase the other two links that you can see have been added in the file.](https://github.com/skillcrush/skillcrush-104/pull/1701/files)

Then, you'll want to add once again `git add -A` and commit `git commit -m "Hope never dies"` and then push up your branch once more: `git push origin ebb-advice` which will update your pull request. 

We're very close!

LenaEbb commented:
Ok....I added the CSS file, fixed the HTML, added and committed, there were no weird files around. Git push gave me error again :((( I'm going to have to watch some tutorials to understand this completely, I feel like walking in the dark. I really want to make this happen but feel stupid at the moment and that doesn't help anyone lol

LenaEbb commented:
I fixed all the files, added and committed, and this is the last I did:
git checkout master
git pull origin master
git merge ebb-advice
git push origin master
Didn't get any errors. What's next? Thanks :)

redrambles commented:
Hey @LenaEbb!

Ok! Thanks for the details of which commands you ran in your terminal. Those aren't quite the commands that we want to run, though - hahaha. But we'll get there!!

Ok! So the first thing is let's fix up what was done on the master branch.

1) Checkout your master branch `git checkout master`.  Then we'll need to rewind back in time to a previous commit before you merged in your branch `git reset --hard beff950`. Remember that we don't want to work directly on the master branch - we just want to merge in the code from the upstream master branch, so that it can stay current. While you are on your master branch, run `git fetch upstream` followed by `git merge upstream/master`. This will bring your local master branch up to date with Skillcrush's master branch.

2) Now let's move back to your topic branch `git checkout ebb-advice`, and let's merge in your freshly updated master branch `git merge master`.  There will be a conflict in your index.html file - that's totally normal! Open it up in your editor and fix it. You'll want to make sure to keep all new student links and remove the conflict markers.

3) Add and commit that change `git add index.html` followed by `git commit -m "bringing up to date with master"` (or similar message). 

4) The last thing to do is to take the css from your ebb-advice18.html advice page and put it in a separate stylesheet (I would recommend that you create a stylesheet and add it the /css directory) - and then make sure to link to this new stylesheet from your ebb-advice18.html page.

5) Add those changes `git add -A` and commit `git commit -m "adding stylesheet and link in index page"`

6) Push up your branch `git push origin ebb-advice`

At that point, things should be looking pretty good, and we'll go from there!! You got this! Let us know if you have questions!

LenaEbb commented:
Hey Ann, I've done everything except removing css from ebb-advice18 and creating another css file, I have done that already yesterday(lena.css). This is what I'm getting after git push origin master 
<img width="560" alt="screen shot 2018-03-23 at 9 20 46 pm" src="https://user-images.githubusercontent.com/35615931/37860359-6fc78324-2ee0-11e8-993c-8b837a6d1fca.png">
I have to admit that I really think that I understand how git and GitHub work and I think I'm learning something every time but then something else surprises me...I guess it takes practice and I hope to really laugh about this someday...haha

redrambles commented:
Hey @LenaEbb ! Ok - thanks for the screenshot!

Do you see how the terminal is informing you that your push was rejected? This is happening because the histories of your local repository and your remote repository have diverged. This usually happens when we make a change directly on GitHub - not realizing that this also creates a new commit in the process. 

My question is -are you happy with how your branch is locally at the moment? If so - you can use something called a 'force' push - but only because this is your branch and nobody else is working on it! :) Basically, by force pushing, we are telling Git to ignore the fact that the histories diverge and to blow away any commits made on the remote copy of the ebb-advice branch that do not exist on your local copy of the same branch. You would do that by running:

`git push -f origin ebb-advice`

This time your push will go through and your pull request will be updated with your changes.

Let us know when this is done and we'll go from there! :)

LenaEbb commented:
Done :) Thanks Ann, I'm realizing that there's more to this than I initially understood. Now this does finally makes sense :)

redrambles commented:
Hey @LenaEbb !

So we're getting closer but we're not there yet. :) 

Here is what remains to be done. In your ebb-advice branch:

1) Delete your `advice/lena-balta-advice` file.

2) Delete your `advice/lenabalta-advice.html` file.

3) Make sure that your `advice/ebb-advice18.html` file links to your css stylesheet properly. Right now it links to the stylesheet as if it were in the root of the project, but it is inside the css directory, so you'll need to edit that path in the link.

4) Add a link to your `advice/ebb-advice18.html` in the index.html file at the root of the project, and delete the remaining conflict markers.

Once you have done this, make sure to save your changes, and then add `git add -A` and commit `git commit -m "Lena's advice"` and push up your branch once more `git push origin ebb-advice` - and we should be good to merge in your code!

You got this!

Ann

LenaEbb commented:
It's done. I hope that'll do it...

acodeinprogress commented:
Hey @LenaEbb,

You're sooo close!

There are just a couple more things that you'll need to do.
1. Your CSS file should be in the css folder rather than advice folder. What you'll want to do is move that file to the right folder. Once you do that, you'll also want to change the link that you have in your advice file to direct to the CSS file inside the css folder rather than in the same folder as your advice file.
2. There are still some merge conflict markers in the index.html file. You'll need to make sure to remove those.

Once you do that, you'll need to run a slightly different git command (git add -A), then commit, and push.

xoxo,
Sharon

LenaEbb commented:
OK, this is officially weird. I actually never noticed before that lena.css wasn't in the css file, so I moved it and fixed the link in my ebb-advice18.html. I went to commit and push and I got the "passphrase needed" or something like that, I never had that before. I tried some passwords but nothing worked, I went to my keychain and tried SHH and there was nothing. I then went to check my files and lena.css was completely deleted from my documents???? I had to go back to GitHub and copy the code and make another file. Then tried git push origin ebb-advice, then git push git checkout master, git push origin master, and it's now again asking for passphrase. So annoying. I did update my OS yesterday, does that have anything to do with it? Also, when I make a change and git add -A, git commit -m "Blah blah", do I have to git push origin "my branch" AND git push origin master? Do I git push from my branch and then move to master and push origin master? Not clear 100% And when do I use pull? I thought I'd be done with this by now and it just got more complicated, is this meant to be this complicated? Thanks for your patience

redrambles commented:
Hey there! You've submitted a new pull request that was accepted, so I'm going to close this one! :)
